src/solvers/coin_cbc.rs

```diff
@@ -166,6 +167,27 @@ impl SolverModel for CoinCbcProblem {
     }
 }
 
+impl WithInitialSolution for CoinCbcProblem {
+    type Solution = CoinCbcSolution;
+
+    fn with_initial_solution(mut self, solution: &Self::Solution) -> Self {
+        self.model.set_initial_solution(&solution.solution);
+        self
+    }
+}
+
+impl FromIterator<(Variable, f64)> for CoinCbcSolution {
+    fn from_iter<T: IntoIterator<Item = (Variable, f64)>>(iter: T) -> Self {
+        CoinCbcSolution {
+            solution: CbcSolution {
+                raw: todo!(),
+                col_solution: todo!(),
+            },
+            solution_vec: iter.into_iter().map(|(_, value)| value).collect(),
+        }
+    }
+}
```

Comment:
@lovasoa I am a bit lost on how to move forward. Did you have something like this in mind? If yes, how can we construct the underlying solution of the solver? The underlying `coin_cbc` crate does not seem to expose any way of creating solutions from scratch. Yet they are the only data type that can be passed to `set_initial_solution`.

Do I first have to go fix `coin_cbc` before this becomes possible?

Reply:
Indeed, I did not have this in mind, and I think this questions our initial approach...

Maybe `with_initial_solution` should instead be generic and take a trait that is implemented both by Solution AND by (var, value) iterators?

Reply:
I'm not really sure how structuring traits differently solves the underlying problem. There is no way to construct an initial solution for `coin_cbc` except by running the solver, so how can it ever work to set initial solutions at all—unless they were obtained from previous runs?

Reply:
I asked about this in https://github.com/KardinalAI/coin_cbc/issues/37

Reply:
I think I'll manage to implement this in a few days thanks to @TeXitoi